openfisca/openfisca-core
# Generate flame chart for performance analysis

Connected to #877

#### New features

- Introduce `openfisca test myTest.yaml --performance`
  - Generates a flame chart in a web page to view the time taken by every calculation in a simulation

To test it:
1. Run a yaml test:
`openfisca test tests/formulas/irpp.yaml --performance -c openfisca_france`
2. Run a Python web server with:
`python -m http.server 5000`
3. See its result in your browser at `http://localhost:5000`

When your yaml file contains multiple tests, only the last one is displayed in the flame chart.
You can use [openfisca test --name_filter option](https://openfisca.org/doc/openfisca-python-api/openfisca-run-test.html) to choose a specific test case.
